### PR TITLE
Improve status badge feature

### PR DESCRIPTION
**Pull request by alexmt**
PR adds missing features of status badge feature:

1. By default, the badge should be unavailable due to security reasons. User should be able to enable the feature using settings.
1. Adds UI which allows generating the badge for markdown, textile etc
1. Improves badge look and feel.

Screen Shot 2019-06-30 at 12 15 35 AM

### CODE REVIEW

File: util/settings/settings.go [hunk 1]
```diff
@@ -112,6 +114,7 @@ const (
 	settingDexConfigKey = "dex.config"
 	// settingsOIDCConfigKey designates the key for OIDC config
 	settingsOIDCConfigKey = "oidc.config"
+	statusBadgeEnabledKey = "statusbadge.enabled"
```
Comment (dthomson25) [inline util/settings/settings.go]: Probably should add a comment similar to all the other constants here.

**Review by dthomson25 — APPROVED** [review]
LGTM other than a minor style comment

**Comment by dthomson25** [comment]
Maybe we should add two e2e tests: one confirming that this feature is enabled and another for when it should be disabled

**Review by alexec — APPROVED** [review]
Rubber-stamp approval.

File: util/settings/settings.go (hunk 1 above)
Comment (alexmt) [reply util/settings/settings.go]: done

**Comment by alexmt** [comment]
Added tests. I've decided to use unit tests instead of e2e since badge functionality does not have much dependencies and can be tested using fake k8s clients.

**Pull request merged**